aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -305,6 +304,9 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
         break;
       }
     }
+  } else {
+    // dst.dim() can be smaller than src.dim() when scalars are copied
+    TORCH_CHECK(dst.numel() == 1, "Destination has ", dst.dim(), " dims, which is fewer than src ", src.dim());
```

Comment:
Shouldn't this check that src.numel() == 1 as well?

Reply:
How about adding `TORCH_CHECK_INTERNAL(src.numel() == dst.numel())` at the start of the function? (but after resize) (On the other hand, check that number of elements for src and dst is the same should already be handled in `Copy.cpp`, shouldn't it?)

Reply:
Did broadcasting already happen by then?

Reply:
Actually is that true?
```
import torch

a = torch.rand(1)
b = torch.rand(1, 1, 1)

a.copy_(b)

```
This fails for me?
Is our cuda copy not consistent with our cpu copy?

Reply:
@albanD well, `output with shape [1] doesn't match the broadcast shape [1, 1, 1]` exception is triggered for `tensor.copy_` method invocation after my change as well, which I guess part of the meta operator call, which item somehow bypasses..